lib/grpfp.gi

```diff
@@ -5866,3 +5866,47 @@ InstallMethod( IndependentGeneratorsOfAbelianGroup,
   IndependentGeneratorsOfMaximalAbelianQuotientOfFpGroup );
 
 BindGlobal( "TRIVIAL_FP_GROUP", FreeGroup(0) / [] );
+
+
+#############################################################################
+##
+#M  ConjugacyClasses( <G> ) . . . . . . . .  for f.p. groups (error messages)
+##
+##  For an FpGroup that does not have the 'IsFinite' flag,
+##  we do not want that calling 'ConjugacyClasses' triggers a call of
+##  'IsFinite',
+##  see the introduction of the Chapter "Finitely Presented Groups"
+##  in the Reference Manual.
+##
+InstallMethod( ConjugacyClasses,
+  "throw an error for f.p. groups without HasIsFinite",
+  [ IsSubgroupFpGroup ],
+  function( G )
+  if HasIsFinite( G ) then
+    if IsFinite( G ) then
+      # Something went wrong, we should not get here.
+      ErrorNoReturn( "there should be a higher ranked method" );
```

Comment:
Isn't this a check that you as developers of GAP have not made a mistake? If so, shouldn't this be an assertion, rather than an error? If I was confronted with the error message, I wouldn't know what I was supposed to do to resolve it.

Reply:
Done.